campaign-launcher/interfaceV2/src/api/httpClient.ts

```diff
@@ -0,0 +1,124 @@
+import axios, { 
+  AxiosError, 
+  AxiosInstance, 
+  AxiosRequestConfig, 
+  AxiosRequestHeaders, 
+  AxiosResponse, 
+  InternalAxiosRequestConfig 
+} from 'axios';
+
+import { tokenManager, TokenData } from '../utils/TokenManager';
+
+type HttpClientConfig = {
+  baseURL: string;
+  headers?: AxiosRequestHeaders;
+};
+
+type RefreshPromise = Promise<AxiosResponse<TokenData>>;
+
+export const REFRESH_FAILURE_EVENT = 'refresh-failure';
+
+export class HttpClient {
```

Comment:
Logic with interceptors and tokens is specific to recording api client, not to this generic http client, so should be moved out of it somehow (probably `RecordingHttpClient extends HttpClient`)